src/stack/Stack.java

```diff
@@ -19,16 +19,21 @@ public Stack(int size) {
 	 * Add element into stack
 	 */
 	public void push(int value) {
-
+		tos = tos + 1;
+		if (tos < data.length) {
```

Comment:
You have to ensure the capacity of the array if it is already full and continue to push elements there.